### DISCUSSION TITLE
Can I show links to previous/next entry?

### DISCUSSION THREAD

**Discussion by Zauberweib** [datenstrom/community#137] [Ask a question] [answered]
I'm missing the navigation from (previous) entry to (next) entry for the blog. 

It's christmas, maybe my wish could be fullfilled? ;)

**Comment by markseuffert** [comment]
Yes, it can be done with some Yellow customisation and the [API](https://datenstrom.se/yellow/help/api-for-developers).

Here's an example plugin code `system/plugins/blogextension.php`:

``` php
<?php
// Copyright (c) 2013-2015 Datenstrom, http://datenstrom.se
// This file may be used and distributed under the terms of the public license.

// Blog plugin
class YellowBlogextension
{
    const Version = "0.6.1";
    var $yellow;            //access to API

    // Handle initialisation
    function onLoad($yellow)
    {
        $this->yellow = $yellow;
    }

    // Handle page content parsing of custom block
    function onParseContentBlock($page, $name, $text, $shortcut)
    {
        $output = NULL;
        if($name=="bloglinks" && $shortcut)
        {
            list($location) = $this->yellow->toolbox->getTextArgs($text);
            if(empty($location)) $location = $this->yellow->config->get("blogLocation");
            $blog = $this->yellow->pages->find($location);
            $pages = $blog ? $blog->getChildren(!$blog->isVisible()) : $this->yellow->pages->clean();
            $pages->filter("template", "blog");
            $page->setLastModified($pages->getModified());
            if(count($pages))
            {
                $pageNumber = -1;
                $pageLocation = $page->getLocation();
                foreach($pages as $key=>$value) if($pageLocation == $value->getLocation()) { $pageNumber = $key; break; }
                if($pageNumber > 0) $pagePrevious = $pages[$pageNumber-1];
                if($pageNumber < count($pages)-1) $pageNext = $pages[$pageNumber+1];
                $output = "<div class=\"".htmlspecialchars($name)."\">\n";
                if($pageNext)
                {
                    $output .= "<a class=\"next\" href=\"".$pageNext->getLocation()."\">← ".$pageNext->getHtml("titleNavigation")."</a>\n";
                }
                if($pagePrevious)
                {
                    $output .= "<a class=\"previous\" href=\"".$pagePrevious->getLocation()."\">".$pagePrevious->getHtml("titleNavigation")." →</a>\n";
                }
                $output .= "</div>\n";
            } else {
                $page->error(500, "Bloglinks '$location' does not exist!");
            }
        }
        return $output;
    }

    // Handle page extra HTML data
    function onExtra($name)
    {
        return $this->onParseContentBlock($this->yellow->page, $name, "", true);
    }
}

$yellow->plugins->register("blogextension", "YellowBlogextension", YellowBlogextension::Version);
?>
```

Here's an example snippet `system/themes/snippets/content-blog.php`: 

``` php
<div class="content main">
...
<?php echo $yellow->page->getExtra("bloglinks") ?>
<?php echo $yellow->page->getExtra("comments") ?>
</div>
```

This will add links to the bottom of each blog page. You can change the look with CSS, at the moment the code will only output the necessary HTML code. It's an example code, please feel free to modify and adjust it to your website. 

Hope this helps, happy coding. :smile:

**Comment by Zauberweib** [comment]
Thanks a lot for your work! Its funny: putting  `<?php echo $yellow->page->getExtra("bloglinks") ?>` in the content-blog.php I only geht "Something went wrong. Bloglinks '/blog/' does not exist!"

Using the shortcut `[bloglinks /]` _in_ the entries, everything is fine and I get the title from the entry before and the following one. 

Any idea? I copied the newest core.php - no changing. Tried it on another intstallation - same message. 

greets & thanks

**Comment by markseuffert** [comment]
Sorry about that, but there should be an easy fix. The function `getExtra()` doesn't accept more arguments, it uses the default arguments for the shortcut. To configure your blog location open file `system/config/config.ini` and add `BlogLocation: /`. 

Does that work?

**Comment by Zauberweib** [comment]
YESS! thats it! you are great, thanks a lot for fullfilling my wish so very fast! tnxtnxtnx!
:)

**Comment by Zauberweib** [comment]
[blogextension.pdf](https://github.com/datenstrom/yellow/files/72867/blogextension.pdf)

Made some little changes; I devided it in two pieces; "blogolder" and "blognewer". And handle it with "span class" instead of "div". 

In the content-blog.php there is:
 `<div class="entry-header"><h1>`
`<?php echo $yellow->page->getExtra("blogolder") ?>`
`<?php echo $yellow->page->getHtml("titleContent") ?>`
`<?php echo $yellow->page->getExtra("blognewer") ?></h1></div>`    

in .css I styled it with float left / float right and now its inside the title of the current entry. 
![blaettermenu](https://cloud.githubusercontent.com/assets/15169591/12012773/4351609a-ad04-11e5-8f99-9cc3a54bfd3e.jpg)

I will show that in zaubsie.bplaced.net too, but your answer was so very fast, so you should see the (first) result fast, too :)

**Comment by markseuffert** [comment]
Good to hear that it works. We can add something like this to the blog plugin later, if many people use it. 

Feel free to close the issue when it works for you.

**Comment by markseuffert** [comment, marked as answer]
Now there's an official plugin: https://github.com/datenstrom/yellow-plugins/tree/master/links

**Reply by cellio** [reply]
That extension is now at https://github.com/datenstrom/yellow-extensions/tree/master/source/previousnext.

**Reply by annaesvensson** [reply]
That extension moved now to https://github.com/annaesvensson/yellow-previousnext
